dhall/src/Dhall/Core.hs

```diff
@@ -2141,3 +2152,49 @@ prettyURIComponent text
 throws :: (Exception e, MonadIO io) => Either e a -> io a
 throws (Left  e) = liftIO (Control.Exception.throwIO e)
 throws (Right r) = return r
+
+{-| Convenience utility identical to @"Control.Lens".`Control.Lens.rewriteOf`@
+    re-exported for convenience in order to minimize dependencies on @lens@
+-}
+rewriteOf :: ASetter a b a b -> (b -> Maybe a) -> a -> b
+rewriteOf l f = go
+  where
+    go = transformOf l (\x -> maybe x go (f x))
+{-# INLINE rewriteOf #-}
+
+{-| Convenience utility identical to @"Control.Lens".`Control.Lens.transformOf`@
+    re-exported for convenience in order to minimize dependencies on @lens@
+-}
+transformOf :: ASetter a b a b -> (b -> b) -> a -> b
+transformOf l f = go
+  where
+    go = f . over l go
+{-# INLINE transformOf #-}
+
+{-| Convenience utility identical to @"Control.Lens".`Control.Lens.rewriteMOf`@
+    re-exported for convenience in order to minimize dependencies on @lens@
+-}
+rewriteMOf
```

Comment:
cc: @ocharles 

I re-exported these utilities along the way, which you might find useful